server/src/test/java/org/apache/druid/server/http/MetadataResourceTest.java

```diff
@@ -0,0 +1,182 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.http;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import org.apache.druid.client.DataSourcesSnapshot;
+import org.apache.druid.client.ImmutableDruidDataSource;
+import org.apache.druid.indexing.overlord.IndexerMetadataStorageCoordinator;
+import org.apache.druid.java.util.common.Intervals;
+import org.apache.druid.metadata.SegmentsMetadataManager;
+import org.apache.druid.server.coordinator.DruidCoordinator;
+import org.apache.druid.server.security.AuthConfig;
+import org.apache.druid.server.security.AuthTestUtils;
+import org.apache.druid.server.security.AuthenticationResult;
+import org.apache.druid.timeline.DataSegment;
+import org.apache.druid.timeline.SegmentStatusInCluster;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+public class MetadataResourceTest
+{
+  private static final String DATASOURCE1 = "datasource1";
+  private static final String DATASOURCE2 = "datasource2";
+
+  private MetadataResource metadataResource;
+
+  private SegmentsMetadataManager segmentsMetadataManager;
+  private DruidCoordinator coordinator;
+  private HttpServletRequest request;
+
+  private final DataSegment dataSegment1 = new DataSegment(
+      DATASOURCE1,
+      Intervals.of("2010-01-01/P1D"),
+      "v0",
+      null,
+      null,
+      null,
+      null,
+      0x9,
+      10
+  );
+
+  private final DataSegment dataSegment2 = new DataSegment(
+      DATASOURCE1,
+      Intervals.of("2010-01-22/P1D"),
+      "v0",
+      null,
+      null,
+      null,
+      null,
+      0x9,
+      20
+  );
+
+  private final DataSegment dataSegment3 = new DataSegment(
+      DATASOURCE2,
+      Intervals.of("2010-01-01/P1M"),
+      "v0",
+      null,
+      null,
+      null,
+      null,
+      0x9,
+      30
+  );
+
+  private final DataSegment dataSegment4 = new DataSegment(
```

Comment:
Nit: see if these can be put in a single line.

Reply:
This does end up being pretty long if I make it a single line, and druid convention is to put all arguments in a new line if it takes more than one line